ProcessOptimizer/space/constraints.py

```diff
@@ -121,6 +124,100 @@ def rvs(self, n_samples=1, random_state=None):
         # We draw more samples when needed so we only return n_samples of the
         # samples
         return rows[:n_samples]
+    
+    def sumequal_sampling(self, n_samples=1, random_state=None):
+        """Draw samples that respect sum_equals constraints.
+
+        The samples are in the original space. They need to be transformed
+        before being passed to a model or minimizer by `space.transform()`.
+
+        Parameters
+        ----------
+        * `n_samples` [int, default=1]:
+            Number of samples to be drawn from the space.
+
+        * `random_state` [int, RandomState instance, or None (default)]:
+            Set random state to something other than None for reproducible
+            results.
+
+        Returns
+        -------
+        * `points`: [list of lists, shape=(n_points, n_dims)]
+           Points sampled from the space.
+        """
+
+        rng = check_random_state(random_state)
+                
+        # Find a point on the plane defined by the sum_equals constraint where
+        # A + B + ... = value. We do this by asking where the diagonal between
+        # the origin and A_max, B_max, ... intersects the plane.
+        d = len(self.sum_equals[0].dimensions)
+        origin = np.zeros(d)
+        delta = np.zeros(d)        
+        for i, dim in enumerate(self.sum_equals[0].dimensions):
+            origin[i] = self.space.bounds[dim][0]
+            delta[i] = self.space.bounds[dim][1] - self.space.bounds[dim][0]
+        A = np.zeros((d,d))
+        B = np.zeros(d)
+        # Row representing the sum constraint
+        A[0,:] = 1
+        B[0] = self.sum_equals[0].value
+        # Rows that define the linear equation for the diagonal along the 
+        # constrained dimensions
+        for i in range(1,d):
+            A[i, i-1] = -delta[i]/delta[0]
+            A[i, i] = 1
+        # Identify the point that lies on the constraint plane and on the diagonal
+        point = np.linalg.solve(A, B)
+        # Use the fact that the vector [1, 1, ...] (a 1 for each constrained 
+        # dimension) is normal to the plane defined by A + B + ... to build
+        # basis-vectors inside the plane, using the null_space function
+        N = matrix(np.ones(d))
+        ns = linalg.null_space(N)
+        # We only need to simulate points up to a distance of half the diagonal
+        # from the origin to A_max, B_max, etc.
+        sim_distance = np.sqrt(np.sum(delta**2)) / 2
```

Comment:
I'm not going to change this right now, since the sampling is still very fast in most cases. I will bundle this change with one that implements a more space-filling sampling in the constraint plane instead of random sampling.

Reply:
If it is fast enough, I see no reason to implement my suggestion. Implementing it would make the code more complicated and harder to read and maintain, which are real costs that shouldn't be taken lightly. Let's table this for now, and accept Morten's code.